### PR TITLE
Amount with progress bar

### PR DESCRIPTION
**Pull request by cnolimit**

### CODE REVIEW

File: src/components/Input.js [hunk 1]
```diff
@@ -3,7 +3,12 @@ import PropTypes from "prop-types";
 import styled from "styled-components";
 
 const SContainer = styled.div`
-  border-bottom: 1px ${props => (props.disabled ? "dashed" : "solid")} #c6c6c6;
+  border-bottom: 1px
```
Comment (IljaDaderko) [inline src/components/Input.js]: This is clean, but a good place to use new styled-components `css` helper for prop based styling if you want. https://www.styled-components.com/docs/api#css

File: demo/src/examples/components/InputMoneyExample.js [hunk 2]
```diff
@@ -38,6 +44,12 @@ const code = `<div>
     prefix="£"
   />
   <InputMoney
+    label='Amount without pennies'
+    placeholder='0'
+    prefix="£"
+    pennies={false}
```
Comment (gomesphoto) [inline demo/src/examples/components/InputMoneyExample.js]: We could just invert the logic to use props in more conventional way, either empty or just pass props

File: src/components/InputMoney.js [hunk 3]
```diff
@@ -36,85 +38,105 @@ class InputMoney extends Component {
...
 
   componentDidMount() {
     if (typeof this.props.getRef === "function") this.props.getRef(this.input);
   }
 
   _onChange = value => {
-    const formatedValue = formatAmount(value);
+    let formatedValue;
+    if (typeof value === "string") {
+      formatedValue = !this.props.pennies
```
Comment (gomesphoto) [inline src/components/InputMoney.js]: instead of `!this.props.pennies` which is just GBP also, we could invert to `this.props.noDecimals`

Comment (gomesphoto) [inline src/components/InputMoney.js]: we could even make it smarter, make the component adjust decimal digits to the currency prefix, from currencies.json `"decimal_digits":2,`

File: src/components/InputAmountProgress.js [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+  transition: all 0.2s ease-in-out;
+  transform-origin: left top;
+  transform: scale(${props => (props.focus || props.content ? 1 : 1.4)})
+    translateY(${props => (props.focus || props.content ? 0 : "15px")});
+  will-change: transform;
+  font-weight: 400;
+`;
+
+class AmountLeftInput extends Component {
+  constructor(props) {
```
Comment (IljaDaderko) [inline src/components/InputAmountProgress.js]: Do you need constructor here?

just 

```
state = {  } // props need to come from this.props
```

**Pull request merged**